src/Fabulous/Runner.fs

```diff
@@ -0,0 +1,116 @@
+// Copyright Fabulous contributors. See LICENSE.md for license.
+namespace Fabulous
+
+open Fabulous.Tracing
+open System
+
+type RunnerDispatch<'msg>()  =
+    let mutable dispatchImpl = (fun (_msg: 'msg) -> failwith "do not call dispatch during initialization" : unit)
+    member x.DispatchViaThunk = id (fun msg -> dispatchImpl msg)
+    member x.SetDispatchThunk v = dispatchImpl <- v
+
+/// Program type captures various aspects of program behavior
+type RunnerDefinition<'arg, 'msg, 'model, 'externalMsg> =
```

Comment:
TBH this should really have always been an interface I think.  I made it a record out of deference to the Elmish/Elm origins

Reply:
The record did not bother me because it's easy to extend with optional and/or custom behaviors to it through `Program.with...`.
An interface could do it too I guess, but might not be as simple to use.
I need to think about it.